### DISCUSSION TITLE
While using a single prisma client for all my files, prisma errors are not caught properly.

### DISCUSSION THREAD

**Discussion by rohitkv090** [prisma/prisma#20473] [Q&A] [answered]
I'm currently working using an approach where I'm using a centralized prisma client.

**db.js**
```js
const { PrismaClient } = require("@prisma/client");
const prisma = new PrismaClient();

module.exports = prisma;
```

when I'm using this approach and using this in my other files API routes try-catch is not able to catch the error properly error is logged but the response is send as 200, but when creating a new prisma client, error are caught and response is not sent. 

**controller.js**
```js
const prisma = require("../db/db");

exports.addProduct = async (req, res) => {
try{
    ...
}catch(error){
  conosle.log(error);
  res.status(500).send({error:error.message});
}
```

```js
const routes = [
  {
    method: "POST",
    url: "/api/product",
    handler: controller.addProduct,
    preHandler:auth
  }]
```

Lets say that error comes in addProduct controller, it will be logged on the console but the response is send as 200 ok.
So is it a fastify error or prisma error.

**Comment by ludralph** [comment]
Hi @kratos-090 👋 

Thank you for raising this question. From the code snippet you shared, it's not clear if this is an issue with either Prisma or Fastify. Can you share a minimal reproduction in a repository so that we can investigate this or share the implementation details of the controller code to provide more visibility?

**Reply by rohitkv090** [reply]
**repository**
https://github.com/kratos-090/prisma-fastify-test

**When using  new client in controller**
<img width="1440" alt="Screenshot 2023-08-01 at 2 00 32 PM" src="https://github.com/prisma/prisma/assets/64844155/a1c8e781-dcf2-4184-b0c8-ab7d574dfd46">


**When using exported prisma client**
<img width="1440" alt="Screenshot 2023-08-01 at 2 00 52 PM" src="https://github.com/prisma/prisma/assets/64844155/2522be3e-7fbf-4cc5-b06d-42d42ad29df5">

as you can see error was logged but the response was send before i think.

**Reply by ludralph** [reply, marked as answer]
Hi @kratos-090, In line 20, of your middleware [file](https://github.com/kratos-090/prisma-fastify-test/blob/b1329979e4a1937caa7ffaf0a28c753efea64b21/src/middleware/prismaMiddelware.js#L20), you are just logging the error to the console. You need to rethrow the error to propagate it back to the caller. You can add this line after line 20
```js
// Rethrow the error to propagate it back to the caller

    throw error
 
```
With this, you should be able to see the error displayed on the client. For example:
<img width="1006" alt="image" src="https://github.com/prisma/prisma/assets/13855608/51eef52d-94c0-4c5c-91f6-0202adad6ed7">

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by rohitkv090** [reply]
yaa its working now, thanks @ludralph
